lms/djangoapps/certificates/signals.py

```diff
@@ -64,3 +65,36 @@ def toggle_self_generated_certs(course_key, course_self_paced):
     """
     course_key = CourseKey.from_string(course_key)
     CertificateGenerationCourseSetting.set_enabled_for_course(course_key, course_self_paced)
+
+
+@receiver(LEARNER_COURSE_PASSING_GRADE, dispatch_uid="new_passing_learner")
+def _listen_for_passing_grade(sender, user, course_id, **kwargs):  # pylint: disable=unused-argument
+    """
+    Listen for a learner passing a course, send cert generation task
+    """
+
+    # No flags enabled
+    if (
+        not waffle.waffle().is_enabled(waffle.SELF_PACED_ONLY) and
+        not waffle.waffle().is_enabled(waffle.INSTRUCTOR_PACED_ONLY)
+    ):
+        return
+
+    # Only SELF_PACED_ONLY flag enabled
+    if waffle.waffle().is_enabled(waffle.SELF_PACED_ONLY):
+        if not courses.get_course_by_id(course_key, depth=0).self_paced:
+            return
+
+    # Only INSTRUCTOR_PACED_ONLY flag enabled
+    elif waffle.waffle().is_enabled(waffle.INSTRUCTOR_PACED_ONLY):
```

Comment:
From an agile development point-of-view, it's better to implement only what's incrementally needed for this ticket.  Since in Phase 1, we're not targeting/testing/etc Instructor-paced courses, changes required for IP should wait until Phase 2.  This way, you can focus on making sure SP works end-to-end before moving onto IP.